spring-cloud-netflix-turbine-stream/src/test/java/org/springframework/cloud/netflix/turbine/stream/TurbineStreamTests.java

```diff
@@ -129,21 +128,17 @@ public void contextLoads() throws Exception {
 		// The message has to be sent after the endpoint is activated, so this is a
 		// convenient place to put it
 		stubTrigger.trigger("metrics");
-		byte[] bytes = new byte[1024];
-		StringBuilder builder = new StringBuilder();
-		int read = 0;
-		while (read >= 0
-				&& StringUtils.countOccurrencesOf(builder.toString(), "\n") < 2) {
-			read = response.getBody().read(bytes, 0, bytes.length);
-			if (read > 0) {
-				latch.countDown();
-				builder.append(new String(bytes, 0, read));
+
+		String responseBody = "";
+		try (BufferedReader buffer = new BufferedReader(
+				new InputStreamReader(response.getBody()))) {
+			for (int i = 0; i <= 3; i++) {
```

Comment:
Not sure i understand why this for loop is only executed 4 times, could you explain?

Reply:
4 calls are required to retreive the entire payload. With 3 calls, we only get

```

event: message
data: {Ping}
```

the 4th call gets you the metrics data line. (discovered that while testing).

Reply:
Could we then just loop through the buffer looking for the specific metrics data line and then break out of the loop?  That way this test is a little less fragile to any future changes.

Reply:
That's reasonable proposition